pkg/origin/metrics.go

```diff
@@ -0,0 +1,62 @@
+package origin
+
+import (
+	"fmt"
+	"reflect"
+	"regexp"
+
+	"github.com/fastly/fastly-exporter/pkg/filter"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// Metrics collects all of the origin inspector Prometheus metrics.
+type Metrics struct {
+	RespBodyBytesTotal   *prometheus.CounterVec
+	RespHeaderBytesTotal *prometheus.CounterVec
+	ResponsesTotal       *prometheus.CounterVec
+	StatusCodeTotal      *prometheus.CounterVec
+	StatusGroupTotal     *prometheus.CounterVec
+	LatencySeconds       *prometheus.HistogramVec
+}
+
+// NewMetrics returns a new set of metrics registered to the registerer.
```

Comment:
Nit: `registerer` 😅  I guess it fits with the -er interface pattern in Go but might I suggest using `registrar`?

Reply:
The comment reflects the type, which is `prometheus.Registerer` – don't shoot the messenger :) and please revert the change.